### ISSUE TITLE
Whitespaces in authorites are retained in Standard Titles

### ISSUE THREAD

**Issue by jenniferward** [rism-digital/muscat#1575]
Whitespaces are (conveniently) stripped from Sources at the start or end of a field (see https://github.com/rism-digital/muscat/issues/1409) but apparently they are retained elsewhere in Muscat. I've noticed it in the Titles. This interferes with the alphabetizing and can also inadvertently lead to duplicates.

Standard titles sorted alphabetically: The ones with whitespaces are at the top.
https://muscat-test.rism.info/admin/standard_titles?clear_filters=true&order=title_asc
![image](https://github.com/rism-digital/muscat/assets/18663291/7077eb87-83f9-48e8-9240-40914a356c3d)

How the first one looks in Edit mode: https://muscat-test.rism.info/admin/standard_titles/50206615/edit
![image](https://github.com/rism-digital/muscat/assets/18663291/53e991d2-f218-4be8-8eba-706b5fe67c28)

We ended up with ` 12 Sonatas` with a whitespace at the beginning:
https://muscat-test.rism.info/admin/standard_titles/5046265
![image](https://github.com/rism-digital/muscat/assets/18663291/357e93e6-8710-4944-9abd-aad2f06ccb3c)

but also the correct `12 Sonatas` 
https://muscat-test.rism.info/admin/standard_titles/3911582
The one with the space at the beginning is showing up in Sources (not stripped, even after saving):
https://muscat-test.rism.info/admin/sources/1001056785/edit
![image](https://github.com/rism-digital/muscat/assets/18663291/8d40e23c-3de1-47f5-8260-783a60efec95)

**Comment by lpugin** [comment]
Additionally, we have about 9,000 duplicates...

**Comment by jenniferward** [comment]
Maybe this is happening in all the non-MARCy areas?

https://muscat-test.rism.info/admin/digital_objects/22934/edit
![image](https://github.com/rism-digital/muscat/assets/18663291/03d93919-a41e-41ae-b2d0-081090c727e6)

https://muscat-test.rism.info/admin/liturgical_feasts/50001171
![image](https://github.com/rism-digital/muscat/assets/18663291/78fe5bc4-0c07-437e-ac50-1ecce430a200)

![image](https://github.com/rism-digital/muscat/assets/18663291/cb29cd02-f1e8-4f6a-b951-90f037a8eedc)

https://muscat-test.rism.info/admin/standard_terms/50001820 
![image](https://github.com/rism-digital/muscat/assets/18663291/045ee700-009c-48fc-94ba-2de4f1342b72)

![image](https://github.com/rism-digital/muscat/assets/18663291/6969c203-dea4-4f69-b76b-74ec76ec67b5)

**Comment by xhero** [comment]
It could be I think they are only stripped in marc, I imagine this is stuff people copy and paste around?

**Comment by jenniferward** [comment]
Yes, 'people' have been known to copy things from anywhere, even from Muscat!

**Comment by xhero** [comment]
I'm looking at this, it seems that the auth files do not strip the input data. I think we need to do it in two steps:

* Fix all identical duplicates, this is a small nightmare but it should be doable
* Automatically strip whitespace to avoid new ones

**Comment by fjorba** [comment]
> I'm looking at this, it seems that the auth files do not strip the input data. I think we need to do it in two steps:

If my experience helps with cases like this one, I think that it is better to first avoid new cases and then fix the old errors. Because otherwise, there is always the chance a newer one pops up after the correction and before the fix is applied. But of course, you know your workflow better.

**Comment by xhero** [comment]
Normally I would do the same! But in this case I don't want to make a fix that automatically strips whitespace on save, and then have problems when editors save old records that might collide with new ones (triggering the unique constraints) if the record is saved again.
In any case fixing the data and updating the system will happen at upgrade time, when the system is offline, so there is no risk of these kind of problems. It is mostly to say that this problem will not be fixed in 11 :)
